File: algorithms/Yandex/algorithms_training_4/warm_up/C++/A.cpp
#include <iostream>
#include <string>

using std::cin, std::cout;


std::string notMin(short* arr, short left, short right) {
    short cur = arr[left];
    for (short i = left + 1; i != right + 1; ++i) {
        if (cur > arr[i]) return std::to_string(cur);
        if (cur < arr[i]) return std::to_string(arr[i]);
    }
    return "NOT FOUND";
}

// компоновщик будет кидать исключение, если в разных файлах будут одинаковые функции
int inputA()  {
    short arr_size;
    short n_req;
    cin >> arr_size >> n_req;

    short* arr {new short[arr_size]};
    for (short i = 0; i != arr_size; ++i) cin >> arr[i];

    short left, right;
    for (short i = 0; i != n_req; ++i) {
        cin >> left >> right;
        cout << notMin(arr, left, right) << '\n';
    }

    delete[] arr;
    return 0;
}

File: algorithms/Yandex/algorithms_training_4/warm_up/Python/A.py
def notMin(arr: list, left:int, right:int):
    cur = arr[left]
    for i in range(left + 1, right + 1):
        if cur > arr[i]:
            return cur
        if cur < arr[i]:
            return arr[i]
    return "NOT FOUND"


arr_size, n_req = map(int, input().split())

arr = list(map(int, input().split()))

for i in range(n_req):
    print(notMin(arr, *map(int, input().split())))
